tests/diagnostics/uninitialized-out-parameters.slang

```diff
@@ -47,5 +47,48 @@ void control_flow_defined(bool b, out float y)
     return;
 }
 
+// Specialized handling for intrinsic asm
+void instrincs_defined(float x, out float exp)
+{
+    __intrinsic_asm "frexp";
+}
+
+// Specialized handling for target switches
+void target_switching_undefined(float x, out float exp)
+{
+    //CHK-DAG: warning 41018: returning without initializing out parameter 'exp'
+    __target_switch {}
+}
+
+void target_switching_defined(float x, out float exp)
+{
+    __target_switch
+    {
+    case glsl: __intrinsic_asm "frexp";
+    }
+}
+
+void target_switching_blocks(uint x, out half v)
+{
+    __target_switch
+    {
+    case hlsl:
+        if ((x & 2) == 0)
+            v = half(1);
+        else
+            v = half(1);
+        return;
+    case glsl:
+    case spirv:
+        {
+            if ((x & 2) == 0)
+                v = half(1);
+            else
+                v = half(1);
+            return;
+        }
+    }
+}
```

Comment:
I am wondering about a case like following,
```
void target_switching_blocks(uint x, out half v)
{
    __target_switch
    {
    case hlsl:
        return; // <=== WOULD IT PRINT THE WARNING WHEN TARGETTING HLSL?
    case glsl:
    case spirv:
        {
            v = half(1);
            return;
        }
    }
}
```

Reply:
I does print the warning, although not exclusively for hlsl targets. The warning here would be returning without initializing `v`

Should we only warn for the particular target? The diagnostics are being evaluated at the IR level so I'm not sure.

Reply:
Aha. I guess, we cannot print the warning if the checking is happening at ir level.